serving/src/main/java/ai/djl/serving/execution/PredictionService.java

```diff
@@ -0,0 +1,52 @@
+package ai.djl.serving.execution;
+
+import ai.djl.inference.Predictor;
+import ai.djl.modality.Input;
+import ai.djl.modality.Output;
+import ai.djl.repository.zoo.ModelNotFoundException;
+import ai.djl.serving.wlm.WorkLoadManager;
+import ai.djl.translate.TranslateException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PredictionService {
+
+    private static final Logger logger = LoggerFactory.getLogger(PredictionService.class);
+
+    private WorkLoadManager executorProvider;
+    private ModelLoadService modelLoadService;
+
+    private Executor modelLoadExecutor = Executors.newSingleThreadExecutor();
+
+    public PredictionService(WorkLoadManager executorProvider, ModelLoadService modelLoadService) {
+        this.executorProvider = executorProvider;
+        this.modelLoadService = modelLoadService;
+    }
+
+    public CompletableFuture<Output> predict(Input input, String modelName)
+            throws ModelNotFoundException {
+        return modelLoadService
+                .getOrLoadModel(input, modelName)
+                .thenApply(model -> model.queued())
+                .thenApplyAsync(
+                        model -> {
+                            model.running();
+                            Predictor predictor = model.getModel().newPredictor();
+                            try {
+
+                                return (Output) predictor.predict(input);
```

Comment:
About Input/Output class,

DJL support any JOPO java class for Translator, and a ZooModel may already bundle default translator, however, we still need Input and Output for DJL serving. As a REST API, we have to handle HTTP input and output. The Input/Output class here is to mapping HTTP request/response to models' input output. I'm trying to define a standard conversion using Input/Output class here.  Ideally, DJL serving can do:  HTTP request -> Input -> Translator POJO in -> NDList in -> NDList out -> Translator POJO out -> Output -> HTTP response. 
In currently implementation, we only support two cases:
1. The model already support Input/Output conversion
2. The Input/Output is a serialization form of NDList.

Reply:
ah yeah exactly how I have understood it.